docs/develop/dapps/payment-processing/accept-payments-in-a-telegram-bot-js.md

```diff
@@ -0,0 +1,440 @@
+    }
+  );
+
+  // Wait until the user enters the number
+  const count = await conversation.form.number();
+```
+Now we calculate the total amount of the order and generate a random string, which we will use to comment on the transaction and add the dumplings postfix
+```js
+  // Get the total cost: multiply the number of portions by the price of the 1 portion
+  const amount = count * 3;
+  // Generate random comment
+  const comment = Math.random().toString(36).substring(2, 8) + "dumplings";
+```
+
+And we save the resulting data to the session so that we can get this data in the next handler.
+```js
+  conversation.session.amount = amount;
+  conversation.session.comment = comment;
+```
+
+We generate links to go to a quick payment and create an inline keyboard
+```js
+const tonhubPaymentLink = generatePaymentLink(
+    process.env.OWNER_WALLET,
+    amount,
+    comment,
+    "tonhub"
+  );
+  const tonkeeperPaymentLink = generatePaymentLink(
+    process.env.OWNER_WALLET,
+    amount,
+    comment,
+    "tonkeeper"
+  );
+
+  const menu = new InlineKeyboard()
+    .url("Click to pay in TonHub", tonhubPaymentLink)
+    .row()
+    .url("Click to pay in TonKeeper", tonkeeperPaymentLink)
+    .row()
+    .text(`I sent ${amount} TON`, "check_transaction");
+```
+And we send our message with the keyboard, where we ask the user to send a transaction to our wallet address with a randomly generated comment
+```js
+  await ctx.reply(
+    `
+Fine, all you have to do is transfer ${amount} TON to the wallet <code>${process.env.OWNER_WALLET}</code> with the comment <code>${comment}</code>.
+
+<i>WARNING: I am currently working on ${process.env.NETWORK}</i>
+
+P.S. You can conveniently make a transfer by clicking on the appropriate button below and confirm the transaction in the offer`,
+    { reply_markup: menu, parse_mode: "HTML" }
+  );
+}
+```
+
+Now all we have to do is create a handler to check for the presence of a transaction
+```js
+export async function checkTransaction(ctx) {
+  await ctx.answerCallbackQuery({
+    text: "Wait a bit, I need to check the availability of your transaction",
+  });
+
+  if (
+    await verifyTransactionExistance(
+      process.env.OWNER_WALLET,
+      ctx.session.amount,
+      ctx.session.comment
+    )
+  ) {
+    const menu = new InlineKeyboard().text("Buy more dumplings🥟", "buy");
+
+    await ctx.reply("Thank you so much. Enjoy your meal!", {
+      reply_markup: menu,
+    });
+
+    // Reset the session data
+    ctx.session.amount = 0;
+    ctx.session.comment = "";
+  } else {
+    await ctx.reply("I didn't receive your transaction, wait a bit");
+  }
+}
+```
+All we are doing here is just checking for a transaction, and provided that it exists, we tell the user about it and reset the data in the session
+
+
+### Bot start
+
+At this moment, we have written all the required code for our bot. If you did everything correctly, it must work when you run it with `npm run app` command in the terminal.
```

Comment:
Instead of `npm run app` inside the paragraph, let's provide user easy-to-use example how to run his app.

## Example

Example:
https://www.tonspace.co/develop/dapps/tutorials/jetton-minter/#step-by-step-guide-1

The only thing I ask for is to make like this:

### Formatting example

To start use command:

```
npm...
```

If you have any problems...